sdk/trace/span.go

```diff
@@ -491,7 +491,7 @@ func (s *span) copyToCappedAttributes(attributes ...attribute.KeyValue) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	for _, a := range attributes {
-		if a.Value.Type() != attribute.INVALID {
+		if a.Value.Type() != attribute.INVALID && a.Key != "" {
```

Comment:
Can we add a comment here explaining these come from the specification? E.g.

```suggestion
		// Ensure attributes conform to the specification:
		// https://github.com/open-telemetry/opentelemetry-specification/blob/v1.0.1/specification/common/common.md#attributes
		if a.Value.Type() != attribute.INVALID && a.Key != "" {
```